src/react/components/pages/homepage/homePage.test.tsx

```diff
@@ -2,91 +2,138 @@ import React from "react";
 import { BrowserRouter as Router } from "react-router-dom";
-import { mount } from "enzyme";
+import { mount, ReactWrapper } from "enzyme";
 import { Link } from "react-router-dom";
 import { IApplicationState, IProject, ITag, IExportFormat, IConnection } from "../../../../models/applicationState";
-import IProjectActions from "../../../../redux/actions/projectActions";
+import IProjectActions, * as projectActions from "../../../../redux/actions/projectActions";
 import CondensedList from "../../common/condensedList/condensedList";
 import FilePicker from "../../common/filePicker/filePicker";
+import MockFactory from "../../../../common/mockFactory";
+import { Store, AnyAction } from "redux";
 
-describe("Connection Picker Component", () => {
-    const defaultState: IApplicationState = initialState;
-    const store = createReduxStore(defaultState);
-    let wrapper: any = null;
-    let recentProjects: IProject[] = null;
-    const actions: IProjectActions = null;
-    let source: IConnection;
-    const tags: ITag[] = [];
-    let format: IExportFormat;
-    const history: any = null;
-    const location: any = null;
-    const match: any = null;
-    let onChangeHandler: (value: any) => void;
-
-    beforeEach(() => {
-        source = {
-            id: "1",
-            name: "connection name",
-            description: "connection description",
-            providerType: "provider",
-            providerOptions: {},
-        };
-        format = {
-            providerType: "provider",
-            providerOptions: {},
-        };
-        recentProjects = [
-            { id: "1",
-              name: "project1",
-              description: "testproject",
-              tags,
-              sourceConnection: source,
-              targetConnection: source,
-              exportFormat: format,
-              autoSave: true,
-            },
-        ];
+jest.mock("../../../../services/projectService");
+import ProjectService from "../../../../services/projectService";
 
-        onChangeHandler = jest.fn();
+describe("Connection Picker Component", () => {
+    const recentProjects = MockFactory.createTestProjects(2);
 
-        wrapper = mount(
+    function createComponent(store, props: IHomepageProps): ReactWrapper<IHomepageProps> {
+        return mount(
             <Provider store={store}>
                 <Router>
-                    <HomePage
-                        recentProjects={recentProjects}
-                        actions={actions}
-                        history={history}
-                        location={location}
-                        match={match}
-                    />
+                    <HomePage {...props} />
                 </Router>
             </Provider>,
         );
-    });
+    }
 
     it("should render a New Project Link", () => {
+        const store = createStore(recentProjects);
+        const props = createProps();
+        const wrapper = createComponent(store, props);
```

Comment:
Is there a reason you are not putting these in a beforeEach method?

Reply:
I can likely move them into a `beforeEach` tag